homu/main.py

```diff
@@ -1239,13 +1233,23 @@ def queue_handler():
             return process_queue(states, repos, repo_cfgs, logger, buildbot_slots, db, git_cfg)
 
     os.environ['GIT_SSH'] = os.path.join(os.path.dirname(__file__), 'git_helper.py')
+    sshcfg = cfg_git.get('ssh_config_path')
+    if sshcfg is not None:
+        os.environ['HOMU_SSH_CONFIG'] = sshcfg
     os.environ['GIT_EDITOR'] = 'cat'
 
+    tmp_ssh_key = None
+    if git_cfg['local_git']:
+        tmp_ssh_key = tempfile.NamedTemporaryFile(prefix='homu-sshkey')
+        tmp_ssh_key.write(git_cfg['ssh_key'].encode('utf-8'))
+        tmp_ssh_key.flush()
+        os.environ['HOMU_GIT_KEY_PATH'] = tmp_ssh_key.name
```

Comment:
This seems like a hack. Passing the filename of the `temp_ssh_key` in an environment variable and then having to add a dummy parameter to `check_timeout` to keep it alive is likely to break down the road. Plus, I've never liked programs changing their environments, which act as superglobals. Instead, we can store `tmp_ssh_key` in `git_cfg`, which is conveniently already threaded through the code into all the places we execute git, thus automatically preserving lifetimes.

`HOMU_GIT_KEY_PATH` still needs to be passed as an environment variable to `git_helper.py`. It turns out (almost!) all calls to git are in one of two forms: `utils.silent_call(git_cmd(...))` or `utils.logged_call(git_cmd(...))`,  and all calls to `utils.silent_call` and `utils.logged_call` invoke git.

I would prefer `git_cmd` (aka the function returned by `init_local_git_cmds`) to return both the complete git command array as well as a dict representing the env vars, e.g. with a tuple or dict.
Then, update `utils.silent_call` and `utils.logged_call` to handle the new interface and pass an updated environment through to `subprocess`.

The same thing should be done for `HOMU_SSH_CONFIG`, `GIT_SSH`, and `GIT_EDITOR`.

Extra fun fact: updating `os.environ` in Python calls `putenv`, which has a flawed interface that [is broken in various ways](https://stackoverflow.com/questions/5873029/questions-about-putenv-and-setenv#5876818). Another reason not to modify it!